.expeditor/scripts/remove_build_channel.sh

```diff
@@ -10,11 +10,11 @@ source .expeditor/scripts/shared.sh
 export HAB_AUTH_TOKEN="${ACCEPTANCE_HAB_AUTH_TOKEN}"
 export HAB_BLDR_URL="${ACCEPTANCE_HAB_BLDR_URL}"
 
-install_latest_hab_binary "$BUILD_PKG_TARGET"
+curlbash_hab "$BUILD_PKG_TARGET"
 
 channel="$(get_release_channel)"
 echo "--- Destroying release channel '${channel}'"
 
-${hab_binary} bldr channel destroy \
+hab bldr channel destroy \
```

Comment:
I would argue that we should continue to use the `hab_binary` variable, even though we don't need it on linux - this way we have consistency on the binlinked vs install.sh based hab binary on OSX. (one installs to /usr/local/bin, the other to /usr/bin IIRC, which is an issue in and of itself, in my opinion)

Reply:
My motivation here was to streamline the code. When you see `${hab_binary}` instead of `hab`, your first thought is "Why are we not using `hab` directly... what is it about this code that means we have to care?" That _is_ a concern in the release pipeline, where we want to ensure that we're building with the Habitat toolchain that we're currently building.

However, in other cases, we really don't care as much; we just need a `hab` binary that will let us do simple things like interact with the Builder API. In those cases, we can just use `curlbash_hab` to install the latest stable `hab`, just like any normal user would, and then just use `hab`. Whether that's in `/usr/bin` or `/usr/local/bin` doesn't matter; the fact that the `install.sh` script that `curlbash_hab` is using under the hood ensures that `hab` is on `$PATH` is the key thing.

Reply:
While I agree on a technical level that we don't have to care in some places, it feels sloppy to rely on the whatever is on the path, especially since we should be using the "newly built hab" for as many things as possible after studio is built, just to exercise it. I my mind the first thing i think when I see `${hab_binary}` is "Ok I can trust this is set correctly" rather than thinking "which hab is on the path right now?" :)

Reply:
Even that won't work because we're pulling from the "current release channel" which is tied to Buildkite job. It's fine in a release pipeline, but will be a no-op in any downstream pipeline.

That is, absent any other changes, the current refactoring actually captures what is really going on now in those downstream pipelines.

Whether or not we want to always use the `hab` binary from `DEV` in those downstream pipelines is probably something to chat about, since I can see various arguments pro and con. Depending on the outcome of that, we can make subsequent changes to more explicitly capture that in our helper functions.

Reply:
That makes sense - I'm totally for splitting up the functions - I think we should probably rename the 2nd one to like "install_release_channel_hab" or even "install_hab_from_channel" with an argument (which would then work for DEV in downstream pipelines.

Reply:
@scotthain Yeah that sounds like a good follow-on refactoring. I'll probably add something like that in my ongoing work on https://github.com/habitat-sh/habitat/pull/6780

Reply:
@scotthain Yup, that sounds good. I'll probably end up adding something like that in my ongoing work for https://github.com/habitat-sh/habitat/pull/6780.